packages/PowerGrid/src/fit/fit-curve.ts

```diff
@@ -0,0 +1,240 @@
+import * as DG from 'datagrok-api/dg';
+import * as grok from 'datagrok-api/grok';
+import {
+  fit,
+  sigmoid,
+  FitErrorModel,
+  FitResult
+} from '@datagrok-libraries/statistics/src/parameter-estimation/fit-curve';
+import {GridColumn} from "datagrok-api/dg";
+import {getChartData} from "./fit-data";
+
+
+export function scaleCoordinates(coordinates: {x: number[], y: number[]}, canvasWidth: number, canvasHeight: number):
+  {x: number[], y: number[]} {
+  const xMin = Math.min(...coordinates.x);
+  const xMax = Math.max(...coordinates.x);
+  const yMin = Math.min(...coordinates.y);
+  const yMax = Math.max(...coordinates.y);
+
+  const xDiff = xMax - xMin;
+  const yDiff = yMax - yMin;
+  const xMid = (xMax + xMin) / 2;
+  const yMid = (yMax + yMin) / 2;
+
+  const coeff = xDiff * canvasHeight >= yDiff * canvasWidth ? canvasWidth / xDiff : canvasHeight / yDiff;
+  const canvasXCenter = canvasWidth / 2;
+  const canvasYCenter = canvasHeight / 2;
+
+  const scaledCoordinates: {x: number[], y: number[]} = {x: [], y: []};
+  for (let i = 0; i < coordinates.x.length; i++) {
+    scaledCoordinates.x[i] = canvasXCenter + coeff * (coordinates.x[i] - xMid);
+    scaledCoordinates.y[i] = canvasYCenter + coeff * (coordinates.y[i] - yMid);
+  }
+
+  return scaledCoordinates;
+}
+
+export function getParams(columns: DG.Column[]): number[] {
+  const coordinates: {x: number[], y: number[]} = {x: columns[0].toList(), y: columns[1].toList()};
+
+  const minY = columns[1].min;
+  const maxY = columns[1].max;
+  const medY = columns[1].stats.med;
+  let xAtMedY = -1;
+  for (let i = 0; i < coordinates.y.length; i++) {
+    if (coordinates.y[i] === medY) {
+      xAtMedY = coordinates.x[i];
+      break;
+    }
+  }
+
+  return [maxY, 1.2, xAtMedY, minY];
+}
+
+
+export class FitCellRenderer extends DG.GridCellRenderer {
+  get name() { return 'fit-old'; }
+
+  get cellType() { return 'fit-old'; }
+
+  get defaultHeight(): number { return 100; }
+
+  get defaultWidth(): number { return 160; }
+
+  getDefaultSize(gridColumn: GridColumn): {width?: number | null, height?: number | null} {
+    return { width: 160, height: 100};
+  }
+
+  // onMouseMove(gridCell: DG.GridCell, e: MouseEvent): void {
+  //   const hitData = onHit(gridCell, e);
+  //   if (hitData.isHit)
+  //     ui.tooltip.show(ui.divV(createTooltip(hitData.cols, hitData.activeColumn, hitData.row)), e.x + 16, e.y + 16);
+  //   else
+  //     ui.tooltip.hide();
+  // }
+
+  onClick(gridCell: DG.GridCell, e: MouseEvent): void {
+    const df = gridCell.grid.dataFrame;
+
+    const dfColumn = df.columns.toList().filter((column) => column.type === DG.COLUMN_TYPE.DATA_FRAME)![0];
+    const coordinateDf = DG.toJs(dfColumn.get(gridCell.cell.row.idx)) as DG.DataFrame;
+    if (!coordinateDf) return;
+
+    const coordinateColumns = coordinateDf.columns.byNames(['x', 'y']);
+    const coordinates: {x: number[], y: number[]} = {x: coordinateColumns[0].toList(),
```

Comment:
Consider using `columns[0].getRawData()` as it will be memory-efficient

Reply:
Do we need coordinates object at all? We could just use two variables.

Reply:
> Do we need coordinates object at all? We could just use two variables.

guess you're right